src/lib/components/composites/input/ChipsInput.svelte

```diff
@@ -0,0 +1,273 @@
+<script lang="ts">
+    import { Label } from "$lib/components/primitives/label";
+    import { Button } from "$lib/components/primitives/button";
+    import CircleAlert from "lucide-svelte/icons/circle-alert";
+
+    interface ChipsInputProps {
+        items: string[];
+        validate: (arg0: string) => boolean;
+        label?: string;
+        labelPosition?: "top" | "left";
+        searchSuggestions?: (arg0: string) => string[];
+        placeholder?: string;
+        resolveAlias?: (arg0: string) => string | undefined;
+    }
+
+    let {
+        items = $bindable(),
+        validate,
+        label,
+        labelPosition = "top",
+        searchSuggestions,
+        placeholder = "",
+        resolveAlias,
+    }: ChipsInputProps = $props();
+
+    /**
+     * Index (from 0 - \<length of items\>) indicating, which chip is currently selected.
+     * Index -1 represent the state, that no chip is selected.
+     */
+    let selectedChipIndex: number = $state(-1);
+    let inputText: string = $state("");
+    // assume input is valid, until the opposite is not proven by a check
+    let isInputValid: boolean = $state(true);
+
+    let suggestions: string[] = $state([]);
+    /**
+     * Index (from 0 - \<length of suggestions\>) indicating, which suggestion is currently selected.
+     * Index -1 represent the state, that no suggestion is selected.
+     */
+    let selectedSuggestionIndex: number = $state(-1);
+
+    /**
+     * Adds the input to the list of items and add a chip to the visualization.
+     *
+     * The input is only added, if it is not a duplicate and not empty (after trimming).
+     * Furthermore, it clears the input.
+     *
+     * @param item the item to add
+     */
+    function addItem(item: string): void {
+        if (item.trim() !== "" && !items.includes(item)) {
+            if (resolveAlias) {
+                const resolvedAlias = resolveAlias(item);
+                if (resolvedAlias == undefined) {
+                    return;
+                }
+                item = resolvedAlias;
+            }
+            items = [...items, item];
+            inputText = "";
+        }
```

Comment:
I think, here's some potential to reduce indentation and make it more readable.
How about:
```suggestion
        const isValidNewItem = item.trim() !== "" && !items.includes(item);
        if (!isValidNewItem) {
            return;
        }

        const resolvedAlias = resolveAlias?.(item);
        if (resolvedAlias) {
            item = resolvedAlias;
        }

        items = [...items, item];
        inputText = "";
```
By negating the conditions to guard clauses, we can return early.

Reply:
Thank you for the hint, you are right.